kit/src/main/java/com/oracle/javafx/scenebuilder/kit/fxom/FXOMDocument.java

```diff
@@ -80,16 +82,37 @@
     
     private List<Class<?>> initialDeclaredClasses;
     
-    public FXOMDocument(String fxmlText, URL location, ClassLoader classLoader, ResourceBundle resources, boolean normalize) throws IOException {
+    /**
+     * Creates a new {@link FXOMDocument} from given FXML source. Depending on the
+     * use case, the {@link FXOMDocumentSwitch} items can be used to configure the
+     * document creation process according to specific needs. E.g. normalization is
+     * not enabled by default, thus if required the {@link FXOMDocumentSwitch}
+     * {@code NORMALIZED} should be added as constructor argument.
+     * 
+     * @param fxmlText    FXML source
+     * @param location    {@link URL} describing the actual document location
+     * @param classLoader {@link ClassLoader} to be used
+     * @param resources   {@link ResourceBundle} to be used
+     * @param switches    {@link FXOMDocumentSwitch} configuration options to enable
+     *                    or disable certain steps in {@link FXOMDocument} creation
+     *                    (e.g. enforcing normalization)
+     * @throws IOException when the fxmlText cannot be loaded
+     */
+    public FXOMDocument(String fxmlText, URL location, ClassLoader classLoader, ResourceBundle resources, FXOMDocumentSwitch... switches) throws IOException {
         this.glue = new GlueDocument(fxmlText);
         this.location = location;
         this.classLoader = classLoader;
         this.resources = resources;
         initialDeclaredClasses = new ArrayList<>();
         if (this.glue.getRootElement() != null) {
+            String fxmlTextToLoad = fxmlText;
+            if (!Set.of(switches).contains(FXOMDocumentSwitch.FOR_PREVIEW)) {
```

Comment:
Extract `Set.of(switches)` to a variable, so you don't create the set twice?

Reply:
There is now on set used for both places where it is required.